tests/testsuite/lints/unused_optional_dependencies/correct_dep_table/mod.rs

```diff
@@ -0,0 +1,42 @@
+use cargo_test_support::prelude::*;
+use cargo_test_support::registry::Package;
+use cargo_test_support::str;
+use cargo_test_support::{file, project};
+
+#[cargo_test(nightly, reason = "edition2024 is not stable")]
+fn case() {
```

Comment:
This should probably be a test in `tests/testsuite/package.rs`, rather than a UI test for lints